client/cmdlfem4x.c

```diff
@@ -303,96 +309,113 @@ int CmdEM410xBrute(const char *Cmd)
 	char filename[FILE_PATH_SIZE]={0};
 	FILE *f = NULL;
 	char buf[11];
-	int i, n, j, binary[4], parity[4];
-
-	char cmdp = param_getchar(Cmd, 0);
+	uint32_t uidcnt = 0;
+	uint8_t stUidBlock = 20;
+	uint8_t *uidBlock = NULL, *p = NULL;
+	int ch;
 	uint8_t uid[5] = {0x00};
-
-	if (cmdp == 'h' || cmdp == 'H') return usage_lf_em410x_sim();
 	/* clock is 64 in EM410x tags */
 	uint8_t clock = 64;
-
-	param_getdec(Cmd,1, &clock);
+	/* default pause time: 1 second */
+	uint32_t delay = 1000;
+	char delaystr[32] = {0x00};
+	
+	char cmdp = param_getchar(Cmd, 0);
+	
+	if (cmdp == 'h' || cmdp == 'H') return usage_lf_em410x_brute();
+	
+
+	cmdp = param_getchar(Cmd, 1);
+	
+	if (cmdp == 'd' || cmdp == 'D') {
+		param_getstr(Cmd,2, delaystr);
+		if (strlen(delaystr) > 0) delay = (uint32_t) atoi(delaystr);
```

Comment:
hm,  dont we have the function **param_get32()**   instead of getting a string?
and as consequence "d 2000",  why not "c 32" for clock?